src/platform/packages/shared/kbn-alerts-ui-shared/src/alert_filter_controls/filter_group.tsx

```diff
@@ -234,14 +230,26 @@ export const FilterGroup = (props: PropsWithChildren<FilterGroupProps>) => {
       next: debouncedFilterUpdates,
     });
 
+    return () => {
+      filterChangedSubscription.current?.unsubscribe();
+    };
+  }, [controlGroup, debouncedFilterUpdates]);
+
+  useEffect(() => {
+    if (controlsUrlState && !urlStateInitialized) {
+      initializeUrlState();
+    }
+
+    if (!controlGroup) {
+      return;
+    }
+
     inputChangedSubscription.current = controlGroup.getInput$().subscribe({
```

Comment:
Same here. I would recommend moving setting up `inputChangedSubscription` into its own use effect so you don't resubscribe when other unrelated depencies change.

```
useEffect(() => {
    if (!controlGroup) {
      return;
    }

    const subscription = controlGroup.getInput$().subscribe({
      next: handleStateUpdates,
    });

    return () => {
      subscription.unsubscribe();
    };
  }, [
    controlGroup,
    handleStateUpdates,
  ]);
```